cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala

```diff
@@ -244,6 +241,25 @@ class SentimentAnalysisSuiteV4 extends TestBase with DataFrameEquality with Text
     .setTextCol("text")
     .setOutputCol("output")
 
+  test("Sentiment Analysis - Include Opinion Mining") {
+    val replies = getDetector
+      .setIncludeOpinionMining(true)
+      .transform(batchedDF)
+      .select("output")
+      .collect()
+    assert(replies(0).schema(0).name == "output")
+    df.printSchema()
+    df.show()
+    val fromRow = SentimentResponseV4.makeFromRowConverter
+
+    replies.foreach { row =>
+
+      val outResponse = fromRow(row.getAs[GenericRowWithSchema]("output"))
+      println("Target text - " + outResponse.result.head.get.sentences.head.opinions.get.head.target.text)
```

Comment:
I want to know if there is a good pattern of asserting nested values

Reply:
Given that the services responses will change, you can just write some simple checks that its within a range of appropriate numbers or something like that. Equality testing is often too brittle for the services